clang/include/clang/Driver/Options.td

```diff
@@ -187,7 +187,8 @@ def opencl_Group : OptionGroup<"<opencl group>">, Group<f_Group>,
                    DocName<"OpenCL options">;
 
 def sycl_Group : OptionGroup<"<SYCL group>">, Group<f_Group>,
```

Comment:
Why is `Group<f_Group>` needed here?
Will `DocName<"SYCL options">` override the `DocName` defined in `Group<f_Group>`

```
def f_Group : OptionGroup<"<f group>">, Group<CompileOnly_Group>,
              DocName<"Target-independent compilation options">;
```

Also,` f_Group` has `CompileOnly_Group`, but some of the options in `sycl_Group` are link time options.

Reply:
The use of `f_Group` here is more in line with existing usage for offloading specific options such as `cuda_Group` and `offload_Group`.  Not all options used are specific to the compilation phase, but are part of the overall compilation to create the device binary.

I checked the generated documentation files, and `sycl_Group` does override `f_Group` in regards to the `DocName`